lib/firmata.js

```diff
@@ -545,6 +545,17 @@ var Board = function(port, options, callback) {
         this.setSamplingInterval(settings.samplingInterval);
       }
       if (settings.skipCapabilities) {
```

Comment:
Yeah I understand the need. The main thing I was reacting to was falling back to an assumption that the board was ATMega328-based. I ran into a similar issue with BreakoutJS a while back and simply created 128 dummy pins and that solved it. Same solution here. The tricky part is the analog pin mapping, but that may actually not be important since the only thing analog pin mapping is useful for (if I remember correctly) is allowing the user to change an analog pin to a digital pin using the analog pin number instead of the digital pin number. You can read and write (PWM) analog values simply using the actual analog pin number. So perhaps a safe assumption is if a user sets skipCapabilities to true, they are also indicating they know the capabilities of the board they are interfacing with and in that case they know if they want to set an analog pin as a digital input or output pin they simply need to use the digital pin number equivalent.